src/legacy/core_plugins/kibana/public/dashboard/panel/dashboard_panel.tsx

```diff
@@ -17,34 +17,68 @@
  * under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import { injectI18n } from '@kbn/i18n/react';
+import { EuiLoadingChart, EuiPanel } from '@elastic/eui';
+import { InjectedIntl, injectI18n } from '@kbn/i18n/react';
 import classNames from 'classnames';
 import _ from 'lodash';
-
-import { PanelHeader } from './panel_header';
+import React from 'react';
+import {
+  ContainerState,
+  Embeddable,
+  EmbeddableFactory,
+  EmbeddableMetadata,
+  EmbeddableState,
+} from 'ui/embeddable';
+import { EmbeddableErrorAction } from '../actions';
+import { PanelState } from '../selectors';
 import { PanelError } from './panel_error';
+import { PanelHeader } from './panel_header';
 
-import {
-  EuiPanel,
-  EuiLoadingChart,
-} from '@elastic/eui';
+export interface DashboardPanelProps {
+  viewOnlyMode: boolean;
+  onPanelFocused?: (panel: string) => {};
+  onPanelBlurred?: (panel: string) => {};
```

Comment:
For `onPanelFocused` and `onPanelBlurred`, the parameters seem to not be panel objects themselves but rather panel indices (or so it seems from how they are called later on in the component).  It may be clearer to be explicit about this here, i.e. `(panelIndex: PanelId) => ...`